src/zenml/integrations/kubernetes/orchestrators/kubernetes_orchestrator.py

```diff
@@ -0,0 +1,436 @@
+)
+from zenml.logger import get_logger
+from zenml.orchestrators import BaseOrchestrator
+from zenml.repository import Repository
+from zenml.stack import StackValidator
+from zenml.utils.docker_utils import get_image_digest
+from zenml.utils.source_utils import get_source_root_path
+
+if TYPE_CHECKING:
+    from zenml.pipelines.base_pipeline import BasePipeline
+    from zenml.runtime_configuration import RuntimeConfiguration
+    from zenml.stack import Stack
+    from zenml.steps import BaseStep
+
+logger = get_logger(__name__)
+
+
+class KubernetesOrchestrator(BaseOrchestrator):
+    """Orchestrator responsible for running pipelines using Kubernetes.
+
+    Attributes:
+        custom_docker_base_image_name: Name of a docker image that should be
+            used as the base for the image that will be run on k8s pods. If no
+            custom image is given, a basic image of the active ZenML version
+            will be used. **Note**: This image needs to have ZenML installed,
+            otherwise the pipeline execution will fail. For that reason, you
+            might want to extend the ZenML docker images found here:
+            https://hub.docker.com/r/zenmldocker/zenml/
+        kubernetes_context: Optional name of a kubernetes context to run
+            pipelines in. If not set, the current active context will be used.
+            You can find the active context by running `kubectl config
+            current-context`.
+        kubernetes_namespace: Name of the kubernetes namespace to be used.
+            If not provided, `default` namespace will be used.
+        synchronous: If `True`, running a pipeline using this orchestrator will
+            block until all steps finished running on Kubernetes.
+        skip_local_validations: If `True`, the local validations will be
+            skipped.
+        skip_cluster_provisioning: If `True`, the k3d cluster provisioning will
+            be skipped.
+    """
+
+    custom_docker_base_image_name: Optional[str] = None
+    kubernetes_context: Optional[str] = None
+    kubernetes_namespace: str = "default"
+    synchronous: bool = False
+    skip_local_validations: bool = False
+    skip_cluster_provisioning: bool = False
+
+    # Class Configuration
+    FLAVOR: ClassVar[str] = KUBERNETES_ORCHESTRATOR_FLAVOR
+
+    @staticmethod
+    def _get_k3d_cluster_name(uuid: UUID) -> str:
+        """Returns the k3d cluster name corresponding to the orchestrator
+        UUID."""
+        # k3d only allows cluster names with up to 32 characters; use the
+        # first 8 chars of the orchestrator UUID as identifier
+        return f"zenml-kubernetes-{str(uuid)[:8]}"
+
+    @staticmethod
+    def _get_k3d_kubernetes_context(uuid: UUID) -> str:
+        """Returns the name of the kubernetes context associated with the k3d
+        cluster managed locally by ZenML corresponding to the orchestrator
+        UUID."""
+        return f"k3d-{KubernetesOrchestrator._get_k3d_cluster_name(uuid)}"
```

Comment:
The UUID Arg could be added to the docstring here

Reply:
True. Haven't touched those functions yet since they are duplicate code from the Kubeflow integration. IMO we should abstract away some of the duplicate code between the different remote orchestrators first so we only need to fix it in one place.

Reply:
Delete this unused code.